synapse/http/proxyagent.py

```diff
@@ -92,6 +87,15 @@ def __init__(
         if bindAddress is not None:
             self._endpoint_kwargs["bindAddress"] = bindAddress
 
+        http_proxy = None
+        https_proxy = None
+        no_proxy = None
+        if use_proxy:
+            proxies = getproxies_environment()
+            http_proxy = proxies["http"].encode() if "http" in proxies else None
+            https_proxy = proxies["https"].encode() if "https" in proxies else None
+            no_proxy = proxies["no"].encode() if "no" in proxies else None
```

Comment:
I know it breaks consistent style a bit, but there's no need to `.encode()` `no_proxy`, as we just `.decode()` later.